### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by Kflash** [candeza/candeza#1]
Candeza is at a stage now where it can be used (*not yet in production*). I need to settle down on a API for the suits before the code get published. That and the list below is holding me back from publish the complete source:

- [ ]  Finish CLI
- [ ] Watch task
- [ ] Support all Harmony features
- [ ] Logging
- [ ] Improve stacktraces and error support
- [ ] Improve preprocessor and reporters
- [ ] Plugins

**Comment by Kflash** [comment]
I finally settled on a API, but the API reveals too much to the end user. It's more important to have informative stacktraces. Thrown error:

```js 
× serial TypeError: Cannot read property 'test' of undefined
    at Object.Suite.test.serial.test [as test] (C:\xampp4\htdocs\candeza\dist\candeza.js:877:140)
```
So after the code is pushed the next goal will be to support same kind of changing as in `AVA` so this ` at Object.Suite.test.serial.test [as test]` will be be reduced to `Object.Suite.serial` for "normal" serial tests.

**Comment by Kflash** [comment]
@TrySound If you want you can try this code now. The basic core is done. Follow the steps below.
This message will be removed shortly.
```js
// install
npm i candeza

var Candeza = require('candeza');
// name on the suite 
var suite = Candeza('My first test');

suite.serial.test('pass', t => {
    t.strictEqual(1, 1);
});
```

If I'm not done by monday with this, I will send you Gitlab link.

**Comment by TrySound** [comment]
@Kflash So you made one more test framework? I thought the main idea is running tests in different environments. Personally I don't need framework. I still want to use mocha or jasmine because I know it, my colleagues know it. Learning one more framework is less time for business tasks. Even if API is similar to something there are many edge cases which are fixed in older stuff.

**Comment by Kflash** [comment]
Basic core yes. The next layer - plugins - will let users use a optional
framework such as jasmine and mocha etc plus a preprocessor. In that way
jasmine will work for node and mocha for browser etc. A preprocessor task
is simply to let users run tests in other  languages then javascript.
A little like karma but will work both for browser and node.

The basic core is just there in case users only use a preprocessor.

In short you choose the tools you need and the language you want and get a
seamless integration.
